pyo3-macros-backend/src/module.rs

```diff
@@ -194,44 +194,78 @@ pub fn pymodule_module_impl(mut module: syn::ItemMod) -> Result<TokenStream> {
 /// module
 pub fn pymodule_function_impl(mut function: syn::ItemFn) -> Result<TokenStream> {
     let options = PyModuleOptions::from_attrs(&mut function.attrs)?;
-    process_functions_in_module(&options, &mut function)?;
+    process_functions_in_module(&mut function)?;
     let krate = get_pyo3_crate(&options.krate);
     let ident = &function.sig.ident;
     let vis = &function.vis;
     let doc = get_doc(&function.attrs, None);
 
     let initialization = module_initialization(options, ident);
-    Ok(quote! {
-        #function
-        #vis mod #ident {
-            #initialization
-        }
+    if function.sig.inputs.len() == 1 {
```

Comment:
I agree this is a nice addition, but we should split this off as a separate PR. The changes to `wrap_pyfunction(_bound)` already have quite a big diff. (And since this is already in a separate commit, it should be easy to split off).

As for the implementation, maybe we can do something similar to `#[pyfunction]` which can already do this. I think [this](https://github.com/PyO3/pyo3/blob/main/pyo3-macros-backend/src/pymethod.rs#L442-L453) is the relevant code for that. This should avoid a lot of this duplication

Reply:
Thanks for the link to the pyfunction code. I will take a look at that and split off the macro changes as a separate PR. I'd prefer to land that one first so that the `wrap_pyfunction_bound` changes can use the single-arg form for pymodule functions. WDYT?

Reply:
I'd be fine with that. The changes are only tangentially related, so the order they land in does not matter to much I think.